packages/bot-skeleton/src/scratch/blocks/Advanced/Loops/controls_for.js

```diff
@@ -72,15 +73,33 @@ Blockly.Blocks.controls_for = {
     },
 };
 
-Blockly.JavaScript.controls_for = block => {
+Blockly.JavaScript.javascriptGenerator.forBlock.controls_for = block => {
     // eslint-disable-next-line no-underscore-dangle
-    const variable0 = Blockly.JavaScript.variableDB_.getName(block.getFieldValue('VAR'), Blockly.Variables.NAME_TYPE);
-    const argument0 = Blockly.JavaScript.valueToCode(block, 'FROM', Blockly.JavaScript.ORDER_ASSIGNMENT) || '0';
-    const argument1 = Blockly.JavaScript.valueToCode(block, 'TO', Blockly.JavaScript.ORDER_ASSIGNMENT) || '0';
-    const increment = Blockly.JavaScript.valueToCode(block, 'BY', Blockly.JavaScript.ORDER_ASSIGNMENT) || '1';
+    const variable0 = Blockly.JavaScript.variableDB_.getName(
+        block.getFieldValue('VAR'),
+        Blockly.Variables.CATEGORY_NAME
+    );
+    const argument0 =
+        Blockly.JavaScript.javascriptGenerator.valueToCode(
+            block,
+            'FROM',
+            Blockly.JavaScript.javascriptGenerator.ORDER_ASSIGNMENT
+        ) || '0';
```

Comment:
There are two things with the same order = `0`. shouldn't the default assignment be unique for each block?

Reply:
This '0' is actually not the default order value but the default return value of the block itself ( we are adding this default values just incase if there no value present in the block and this '0' will never affect the order ).